community/radare2.hpkg

```diff
@@ -0,0 +1,105 @@
+                (join-pkg-paths ":" "/lib64/pkgconfig"
+                                [libuv libzip])
+                ":"
+                (join-pkg-paths ":" "/lib/pkgconfig"
+                                [capstone zlib openssl xxhash lz4])))
+
+    (os/setenv "PATH"
+               (join-pkg-paths ":" "/bin"
+                               [core/coreutils
+                                core/findutils
+                                core/gcc
+                                core/grep
+                                core/make
+                                core/patch
+                                core/sed
+                                core/awk
+                                core/tar
+                                core/pkgconf
+                               ]))
+    (core/link-/bin/sh)
+    (unpack-src radare2-src)
+    (os/setenv "CFLAGS"
+               (string *default-cflags*
+                       " "
+                       (string/join (map |(string "-I" ($ :path) "/include")
+                                         [capstone
+                                          file
+                                          zlib
+                                          lz4
+                                          xxhash
+                                          openssl
+                                          libuv
+                                          libzip]) " ")
+                       " "
+                       (sh/$<_ pkg-config --cflags capstone "liblz4" zlib libuv libxxhash openssl)))
+    (os/setenv "LDFLAGS"
+               (string *default-ldflags*
+                       " "
+                       (string/join (map |(string "-L" ($ :path) "/lib")
+                                         [capstone
+                                          file
+                                          zlib
+                                          lz4
+                                          xxhash
+                                          openssl]) " ")
+                       " "
+                       (string/join (map |(string "-L" ($ :path) "/lib64")
+                                         [libuv
+                                          libzip]) " ")
+                       " "
+                       "-Wl,-rpath=" (string
+                                      (dyn :pkg-out) "/lib" # to find own libs
+                                      ":"
+                                      (join-pkg-paths ":" "/lib"
+                                                      [capstone file zlib lz4 xxhash openssl])
+                                      ":"
+                                      (join-pkg-paths ":" "/lib64"
+                                                      [libuv libzip]))
+                       " "
+                       "-lmagic" # doesn't have a .pc
+                       " "
+                       (sh/$<_ pkg-config --libs capstone zlib "liblz4" libxxhash openssl libuv libzip)))
```

Comment:
To get openssl stuff used, I think one needs to do things a bit differently: https://github.com/andrewchambers/hpkgs/blob/master/community/python3.hpkg#L45

Specifically, use `libssl` and `libcrypto`, not `openssl`.  This only applies to `pkg-config` invocations in this file IIUC (so also line 61).

It may be obvious, but I think for pkg-config we're asking about specific libraries (of which there can be more than one per "package" -- in this case, openssl provides more than one library and none of them is called `openssl`).

I've found this can get confusing because often enough the package name and the library name match, but this is may be worth noting as a potential gotcha.

Reply:
I don't know if there was error output from using `openssl` (I didn't notice, but then I didn't look too closely either).

Reply:
I found this in the pkg-config man page:
> --print-errors
> If one or more of the modules on the command line, or 
> their dependencies, are not found, or if an error occurs in 
> parsing a .pc file, then this option will cause errors  
> explaining the problem to be printed.

May be this is worth considering for all .hpkg files that use `pkg-config`.

Reply:
I see what you mean, but I think it did work, because there's explicitly `-I` and `-L` for openssl, whereas the `-l` have been added by the `pkg-config openssl --[libs|cflags]`. Example, on my machine:
`pkg-config openssl --cflags --print-errors # result: empty`
`pkg-config openssl --libs --print-errors # result: -lssl -lcrypto`
or with the hpkg:
```
PKG_CONFIG_PATH=/hpkg/<hash>-openssl/lib/pkgconfig pkg-config openssl --cflags
-I/hpkg/<hash>-openssl/include 

PKG_CONFIG_PATH=/hpkg/<hash>-openssl/lib/pkgconfig pkg-config openssl --libs  
-L/hpkg/<hash>-openssl/lib -lssl -lcrypto 

PKG_CONFIG_PATH=/hpkg/<hash>-openssl/lib/pkgconfig pkg-config libssl --libs
-L/hpkg/<hash>-openssl/lib -lssl 

PKG_CONFIG_PATH=/hpkg/<hash>-openssl/lib/pkgconfig pkg-config libcrypto --libs
-L/hpkg/<hash>-openssl/lib -lcrypto 
```
Nevertheless, it might be better to do it like in `python3.hpkg` because then we can drop the explicit `-I`s and `-L`s.

Reply:
Ah, sorry.  I was mistaken.

From the pkg-config man page:

>  The package name specified on the pkg-config command line is defined to be the name of the metadata file, minus the .pc extension.

Here, doing `ls /hpkg/*openssl/lib/pkgconfig/` yields:
```
libcrypto.pc  libssl.pc  openssl.pc
```

It does make sense to use libcrypto and libssl to get the flags we are after.  However, if we were to also use `LDLIBS`, using `openssl` with `pkg-config --libs-only-l`, we'd get just the `-l` info for libcrypto and libssl IIUC.  If we did that though, we'd probably want to do use `pkg-config --libs-only-L --libs-only-other` for `LDFLAGS`.

This kind of detailed splitting seems to matter for some packages and I wasn't aware of it until very recently.  It's not done yet, but here's an example I'm working on that has an example: https://gist.github.com/sogaiu/fbaada1fa249c5dc013b7913e56715df

I think I need to study the details of .pc files more :)

Reply:
If we don't need to use LDLIBS for this case, let's not do so.  We can modify it if necessary later.

Reply:
Then I am leaving it as is, here, IIUC?

Reply:
Yes, I think for the moment that makes sense, thanks!

(This was the first time I tried github's code review and I think it was kind of confusing -- how did you find it?)

Reply:
I agree with you somewhat. I think I should've polished the packages a bit more before I opened the PR, because with this many comments, it gets hard to follow. But also perhaps I should've opened separate PRs for each package. That would've focused the subject matter of the PR and probably make it easier to discuss and reason about. Do you think it would be a good idea to recommend 1 package=1 PR? Or other thoughts?

Reply:
I think the one package file per PR thing is probably worth trying.

I say "file" because some .hpkg files have more than one package defintion (actually most have at least two because defsrc is actually defining a package too).

Reply:
Having said that, for this PR, I'm ok with it remaining in its current state.

Reply:
Good point. Agreed.